patternfly/patternfly-elements
# feat: new (pfe-readtime) to display the estimated time to read

## <Pull request description>
New web component for calculating the read time of a section of a page. This can be done by getting a value inside a data attribute or by capturing the number of words in a given section and dividing it by the read rate.


New web component for calculating the read time of a section of a page.

### Preview


Link(s) to demo page(s) where this element can be viewed:
- [Link](https://deploy-preview-1510--happy-galileo-ea79c4.netlify.app/elements/pfe-readtime/demo/) 


### Testing instructions



1. If you add a `word-count` data attribute to `pfe-readtime` it should use that to calculate the readtime.
2. Example: word-count="250" should result in a readtime of 1-minute
3. If you do not add a `word-count` attribute to `pfe-readtime` it should count the words in the section and do the calculation based on that.
4. If the readtime is less than 1 minute the pfe-readtime should not render on the page.
5. If you change the language code to zh it should change the wpm readtime to 158.
6. If you change word-count="158" on english it should not show the readtime. When you change the language code to `zh` it should say 1-minute readtime.


#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist





### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
